src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    Animal.alive = Animal.alive.map(element => {
+      if (element.name === this.name) {
+        element.hidden = true;
+        this.hidden = true;
+
+        return element;
+      }
+
+      return element;
+    });
+  }
```

Comment:
This is an instance, so no need to use map here, you can simply change this.hidden to true
```suggestion
    this.hidden = true;
```